david-martin/example-prometheus-nodejs
# Add OpenShift template with monitoring CRs

For use with https://github.com/integr8ly/application-monitoring-operator.

The application-monitoring-operator will use this repo as an example project in OpenShift that has the necessary Custom Resources (CRs) for the monitoring stack to reconcile into Alerts, Scape configs & Grafana dashboards.

To verify:

* Install the monitoring stack by following instructions in https://github.com/integr8ly/application-monitoring-operator#installation
* Install this app to a new namespace/project
    * `oc new-project example-prometheus-nodejs`
    * `oc patch ns example-prometheus-nodejs --patch '{"metadata":{"labels":{"monitoring-key":"middleware"}}}'`
    * `oc patch ns application-monitoring --patch '{"metadata":{"labels":{"monitoring-key":"middleware"}}}'`
    * `oc process -f template.yaml | oc create -f -`
* Verify alerts, targets & dashboard in Promethues & Grafana

![image](https://user-images.githubusercontent.com/878251/52225301-fcf0dc80-28a1-11e9-8544-8450ef427ab7.png)
![image](https://user-images.githubusercontent.com/878251/52225310-01b59080-28a2-11e9-8d06-c1891cb8f50e.png)
![image](https://user-images.githubusercontent.com/878251/52225356-15f98d80-28a2-11e9-8256-8daf0c75d4ad.png)
